### PR TITLE
[tests] Add test to verify that all our public APIs are documented.

### PR DESCRIPTION
**Pull request by rolfbjarne**
Of course all our public APIs aren't documented, so there's a *huge* list
of known failures.

But at least this way we won't add more undocumented APIs by accident
(although new APIs will show up every time we add more bindings, in particular
in Xcode season).

### CODE REVIEW

File: tests/cecil-tests/Documentation.cs [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+				if (xmlMembers.TryGetValue (dll, out var xml)) {
+					xml.PubliclyVisible = dll.PubliclyVisible;
+				}
+			}
+
+			var documentedButNotPresent = xmlMembers.Except (dllMembers).ToList ();
+			Assert.Multiple (() => {
+				foreach (var doc in documentedButNotPresent)
+					Assert.Fail ($"{doc}: Documented API not found in the platform assembly. This probably indicates that the code to compute the doc name for a given member is incorrect.");
+			});
```
Comment (mandel-macaque) [inline tests/cecil-tests/Documentation.cs:57]: Is it intended that we fail in the Multiple fails or do we want to add the following Assert.Fails in this multiple assert?

**Review by mandel-macaque — APPROVED** [review]
Minor comment.

File: tests/cecil-tests/Documentation.cs [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+			var rv = new HashSet<AssemblyApi> ();
+			var doc = new XmlDocument ();
+			doc.LoadWithoutNetworkAccess (xml);
+			foreach (XmlNode node in doc.SelectNodes ("/doc/members/member")!) {
+				rv.Add (new AssemblyApi (node.Attributes! ["name"]!.Value!, null));
+			}
+			return rv;
+		}
+
+		class AssemblyApi {
```
Comment (haritha-mohan) [inline tests/cecil-tests/Documentation.cs:102]: Curious on why System.Reflection was used here over Roslyn?

File: tests/cecil-tests/Documentation.cs [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+				name += "~" + GetDocId (md.ReturnType);
+			}
+
+			return name;
+		}
+
+		static string GetDocId (EventDefinition ed)
+		{
+			return GetDocId (ed.DeclaringType) + "." + ed.Name;
+		}
```
Comment (haritha-mohan) [inline tests/cecil-tests/Documentation.cs:171]: nit slightly cleaner syntax

```suggestion
		static string GetDocId (EventDefinition ed) => GetDocId (ed.DeclaringType) + "." + ed.Name;
```

**Pull request merged**